### PR TITLE
fix: convert DAPR_HEALTH_TIMEOUT to float

### LINKED ISSUE #688: [BUG] DAPR_HEALTH_TIMEOUT as environment variable is not converted to numerical value
## Expected Behavior

When passing `DAPR_HEALTH_TIMEOUT` as environment variable it should be converted to a numerical value.

## Actual Behavior

docker container with env `DAPR_HEALTH_TIMEOUT=30` 

```timestamp="2024-03-22T11:03:26" level=ERROR message="Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/starlette/routing.py", line 705, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/usr/local/lib/python3.11/contextlib.py", line 210, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/app/proxy/proxy_app.py", line 35, in lifespan
    DaprClient()
  File "/usr/local/lib/python3.11/site-packages/dapr/clients/__init__.py", line 81, in __init__
    super().__init__(address, interceptors, max_grpc_message_length)
  File "/usr/local/lib/python3.11/site-packages/dapr/clients/grpc/client.py", line 131, in __init__
    DaprHealth.wait_until_ready()
  File "/usr/local/lib/python3.11/site-packages/dapr/clients/health.py", line 45, in wait_until_ready
    remaining = (start + timeout) - time.time()
                 ~~~~~~^~~~~~~~~
TypeError: unsupported operand type(s) for +: 'float' and 'str'
```

## Steps to Reproduce the Problem

Start a docker container with an example env variable value `DAPR_HEALTH_TIMEOUT=30` and initialize `DaprClient()`

### PR DESCRIPTION
**Pull request by KentHsu**
# Description

Convert settings.DAPR_HEALTH_TIMEOUT from str to float
Update DaprHealthCheckTests mock so the test can catch this error

## Issue reference

Fixes #688 [BUG] DAPR_HEALTH_TIMEOUT as environment variable is not converted to numerical value

Please reference the issue this PR will close: #_[#688]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Review by elena-kolevska — APPROVED** [review]

File: tests/clients/test_heatlhcheck.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def test_wait_until_ready_success_with_api_token(self, mock_urlopen):
         self.assertIn('Dapr-api-token', headers)
         self.assertEqual(headers['Dapr-api-token'], 'mytoken')
 
-    @patch.object(settings, 'DAPR_HEALTH_TIMEOUT', 2)
+    @patch.object(settings, 'DAPR_HEALTH_TIMEOUT', '2')
```
Comment (elena-kolevska) [inline tests/clients/test_heatlhcheck.py:65]: Let's make this 2.5, so it's more explicit.
```suggestion
    @patch.object(settings, 'DAPR_HEALTH_TIMEOUT', '2.5')
```

**Comment by elena-kolevska** [comment]
Thanks for the contribution @KentHsu, and for reporting @PhilBug.
This problem is fixed in [this PR](https://github.com/dapr/python-sdk/pull/679/files#diff-bcd316f26f7228d016d43b5d1d6db67ecbad464c33efc674704db907ba26effb), but that would make the value an int, not a float. 

Just a small nit about updating the test to a float value, so it's more explicit.

**Review by elena-kolevska — CHANGES_REQUESTED** [review]
Thanks for the contribution @KentHsu, and for reporting @PhilBug.
This problem is fixed in [this PR](https://github.com/dapr/python-sdk/pull/679/files#diff-bcd316f26f7228d016d43b5d1d6db67ecbad464c33efc674704db907ba26effb), but that would make the value an int, not a float. 

Just a small nit about updating the test to a float value, so it's more explicit.

**Comment by KentHsu** [comment]
Thank you for reviewing this PR @elena-kolevska 
I update the timeout value to 2.5 and also update the assertion to match this change.
Please let me know if there's anything you think we need to update. Thanks!

File: tests/clients/test_heatlhcheck.py (hunk 1 above)
Comment (KentHsu) [reply tests/clients/test_heatlhcheck.py]: updated in [this commit](https://github.com/dapr/python-sdk/pull/692/commits/7903e88712effe4f4d893b95a47182345d9f1373)

**Review by elena-kolevska — APPROVED** [review]

**Review by berndverst — APPROVED** [review]

**Pull request merged**